wandb/sdk/wandb_config.py

```diff
@@ -227,7 +227,11 @@ def _sanitize(self, key, val, allow_val_change=None):
             allow_val_change = True
         # We always normalize keys by stripping '-'
         key = key.strip("-")
-        val = json_friendly_val(val)
+        if not (
```

Comment:
This  is confusing to me - you have instrumented the `json_friendly` code path to handle Artifacts, but are explicitly removing them here? Won't this create JSON encoding issues later on?

Reply:
We want to keep the value in the config as the artifact itself, and not the json encoded version. That way users can use the artifact in the config

Reply:
So I'm having the interface convert it prior to sending to the backend process